torchvision/io/video_reader.py

```diff
@@ -72,7 +72,11 @@ class VideoReader:
 
     Args:
 
-        path (string): Path to the video file in supported format
+        src (string, bytes object, or tensor): The media source.
```

Comment:
Possibly it is a bad idea to change this, despite being a beta API, but leaving it as path because misleading too.. I am open for suggestions from the reviewers

Reply:
Since our Beta API has no warning (which make few users aware of it), I think we should follow deprecation policy for BC changes. I suggest we follow similar approach to transform to deprecate `resample` param: https://github.com/pytorch/vision/blob/21790df9a4f77bd9ec4db44de04594cb539457a7/torchvision/transforms/transforms.py#L1275-L1278

Basically we allow both `path` and `src` (put `path` as optional and the last param), and give warning for those who use `path` that it will be removed on v 0.17.

Reply:
But in that case if someone is using  `VideoReader(path="bla.avi")` wont be able to do it. So I am thinking that we should do the other way around, i.e., we add `src` as optional for now and in two releases it replaces `path`. lmk your thoughts

Reply:
Actually I don't see any good option

**Option 1: Make path optional**
This is non BC because you no longer can do `VideoReader(path="bla.avi")`. 

**Option 2: Make src optional**
This option is BC and allows users to do `VideoReader(path="bla.avi")`, `VideoReader("bla.avi")` and `VideoReader(bytesobj)` but if one wants to use the src keyword then it looks weird: `VideoReader("", src="bla.avi")`. Furthermore later wen we deprecate path it will be a BC change



**Option 3: add an extra argument permanently**
Third option  will be to add `video_obj` as last optional argument, and when this was provided then we we would ignore the `path`. This is BC but a bit ugly when we want to construct a VideoReader from memory : `VideoReader("", video_obj=bytesobj)`

Given this I am inclined for option 3, unless I am missing something

Reply:
My thinking is similar with Option 2, but we let path to be optional at the end, and we set default value of src as empty string.
```
VideoReader(src: Any = "", ..., path: Optional[str] = None)
```

and inside constructor we can handle like:
```
if src == "":
  if path is None:
    raise ValueError(...)
  src = path
```

what do you think?

Reply:
yeah that makes a lot of  sense.  Thanks